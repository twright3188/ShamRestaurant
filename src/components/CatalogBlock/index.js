import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/Ionicons';
import { StyleSheet, TouchableOpacity,Image } from 'react-native';
import Colors from '../../theme';
import GreenTitle from '../GreenTitle';
import FastImage from 'react-native-fast-image'

const styles = StyleSheet.create({
  blockStyles: {
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
    height: 75,
    paddingLeft: 23,
    paddingRight: 100,
    borderBottomWidth: 1,
    borderBottomColor: Colors.skeleton
  },
  titleStyles: {
    color: Colors.catalogListItem
  }
});

function CatalogBlock(props) {
  const { children, icon, last, onPress, image } = props;

  return (
    <TouchableOpacity style={[styles.blockStyles, last ? { borderBottomWidth: 0 } : null]} onPress={onPress}>
      <GreenTitle fontSize={20} textColor={Colors.catalogListItem}>
        {children}
        {icon ? ' ðŸ”¥' : null}
      </GreenTitle>
     <FastImage style={{height:70,width:70}} source={{uri:image}} resizeMode='contain'/>
    </TouchableOpacity>
  );
}

CatalogBlock.propTypes = {
  children: PropTypes.node,
  icon: PropTypes.bool,
  last: PropTypes.bool,
  onPress: PropTypes.func,
};

CatalogBlock.defaultProps = {
  children: '',
  icon: false,
  last: false,
  onPress: null
};

export default CatalogBlock;