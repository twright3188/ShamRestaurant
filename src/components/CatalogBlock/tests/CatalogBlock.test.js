import React from 'react';
import renderer from 'react-test-renderer';
import CatalogBlock from '../index';

it('catalog block renders correctly', () => {
    const tree = renderer.create(<CatalogBlock icon last onPress={() => {}}>Spring</CatalogBlock>).toJSON();
    expect(tree).toMatchSnapshot();
});