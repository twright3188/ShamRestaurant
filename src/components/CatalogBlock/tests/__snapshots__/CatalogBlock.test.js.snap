// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catalog block renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#F2F7F4",
      "borderBottomWidth": 0,
      "flexDirection": "row",
      "height": 72,
      "opacity": 1,
      "paddingLeft": 23,
      "paddingRight": 33,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignItems": "center",
          "color": "#399AE5",
          "fontFamily": "Helvetica Neue",
          "fontWeight": "bold",
          "justifyContent": "center",
          "position": "relative",
          "width": "100%",
        },
        Object {
          "fontSize": 20,
          "lineHeight": 24,
        },
        Object {},
      ]
    }
  >
    Spring
     🔥
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#C8CCC9",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
